engine/src/image_rep.h

```diff
@@ -77,11 +77,17 @@ class MCImageRep
     // MERG-2014-09-16: [[ ImageMetadata ]] Support for image metadata property
     virtual bool GetMetadata(MCImageMetadata& r_metadata) = 0;
     
+    bool Lock();
```

Comment:
I think 'IsLocked()' should be:
```
  virtual bool IsLocked(void) const;
```
With the default implementation returning false (as non-mutable image reps are immutable and so locking makes no sense).

Then in the mutable imagerep subclass, it should be overriden to depend on a flag which is controlled by Lock/Unlock(). As the latter two are only ever called *by* the mutable image rep - they don't need to be anywhere other than in that class.